src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncTest.java

```diff
@@ -187,15 +190,7 @@ void sync_WithErrorUpdatingAndTryingToRecoverWithFetchException_ShouldApplyError
         final TaxCategorySync sync = new TaxCategorySync(options, taxCategoryService);
         final TaxCategoryDraft draft = TaxCategoryDraftBuilder.of("someName", emptyList(), "changed")
             .key("someKey").build();
-        final TaxCategory taxCategory = mock(TaxCategory.class);
-
-        when(taxCategory.getKey()).thenReturn("someKey");
-
-        when(taxCategoryService.fetchMatchingTaxCategoriesByKeys(any()))
-            .thenReturn(completedFuture(new HashSet<>(singletonList(taxCategory))));
-        when(taxCategoryService.updateTaxCategory(any(), any())).thenReturn(supplyAsync(() -> {
-            throw new io.sphere.sdk.client.ConcurrentModificationException();
-        }));
+        buildTaxCategoryWithConcurrentModificationUpdate();
         when(taxCategoryService.fetchTaxCategory(any())).thenReturn(supplyAsync(() -> {
```

Comment:
This method `sync_WithErrorUpdatingAndTryingToRecoverWithFetchException_ShouldApplyErrorCallbackAndIncrementFailed`
seems duplicate with the test moved from IT. Since IT test cases have better assertion and verification, i suggest to remove this method

Reply:
https://github.com/commercetools/commercetools-sync-java/commit/35354acd7fb642d6d63bd95117114b1479db687d